common.py

```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/env python
+
+import os
+import logging
+import tarfile
+import urllib
+import subprocess
+import json
+import time
+import errno
+
+
+logging.basicConfig(level=logging.INFO)
+log = logging.getLogger("common")
+
+packet_auth_token = os.environ["PACKET_AUTH_TOKEN"]
+packet_project_id = 'b4357aca-d706-4096-9e44-0a7c69d40ebe'
+packet_machine_type = 'baremetal_1'
+packet_os_img = "ubuntu_14_04"
+packet_cloud_config= os.environ["PACKET_CLOUD_CONFIG"]
+
+gce_key_file_content = os.environ["GCE_SERVICE_KEY_JSON"]
+gce_key_file = "servicekey.json"
+gce_machine_type = "n1-standard-4"
+gce_os_img_family = "ubuntu-1510"
+gce_project = "rancher-dev"
+gce_zone = 'us-central1-f'
+gce_startup_script_rancher = os.environ["GCE_STARTUP_SCRIPT_RANCHER"]
+gce_startup_script_nfs = os.environ["GCE_STARTUP_SCRIPT_NFS"]
+
+nfs_name = os.environ["GCE_NFS_SERVER_NAME"]
+rancher_name = os.environ["GCE_RANCHER_SERVER_NAME"]
+packet_host_names = os.environ["PACKET_HOST_NAMES"].split(",")
+
+
+def untar(fname):
+    tar = tarfile.open(fname)
+    tar.extractall()
+    tar.close()
+    log.info("Extracted to Current Directory")
+
+
+def install_gcloud(gcloud_sdk_url, gcloud_sdk_file_name):
+    urllib.urlretrieve(gcloud_sdk_url, gcloud_sdk_file_name)
+
+    # unzip it
+    untar(gcloud_sdk_file_name)
+
+    # set env to disable the prompt and run installation script
+    os.environ["CLOUDSDK_CORE_DISABLE_PROMPTS"]="1"
+    retCode = subprocess.call("./google-cloud-sdk/install.sh", shell=True)
+    if retCode != 0:
+        raise Exception("failed to google-cloud-sdk/install.sh")
+    retCode = subprocess.call("./google-cloud-sdk/bin/gcloud auth activate-service-account --key-file " + gce_key_file, shell=True)
+    if retCode != 0:
+        raise Exception("failed to gcloud auth activate-service-account")
+    retCode = subprocess.call("./google-cloud-sdk/bin/gcloud config set project " + gce_project, shell=True)
+    if retCode != 0:
+        raise Exception("failed to gcloud config set project")
+    retCode = subprocess.call("./google-cloud-sdk/bin/gcloud components update", shell=True)
+    if retCode != 0:
+        raise Exception("failed to gcloud components update")
+
+
+def gce_wait_for_operation(compute, operation):
+    log.info('Waiting for GCE operation to finish...')
+    while True:
+        result = compute.zoneOperations().get(
+            project=gce_project,
+            zone=gce_zone,
+            operation=operation).execute()
+
+        if result['status'] == 'DONE':
+            if 'error' in result:
+                raise Exception(result['error'])
+            return result
+
+        time.sleep(10)
+
+
+def install_python_client(python_client):
+    # install pip first, must be root
+    import getpass
```

Comment:
ran the flake8 and it does not complain import at function level. I already got rid of this case "import getpass" here anyway.